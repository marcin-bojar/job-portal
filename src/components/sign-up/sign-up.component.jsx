import React from 'react';
import { connect } from 'react-redux';

import FormInput from '../form-input/form-input.component';
import CustomButton from '../custom-button/custom-button.component';

import { signUpStart } from '../../redux/user/user.actions';

import './sign-up.styles.scss';

export class SignUp extends React.Component {
  constructor() {
    super();

    this.state = {
      displayName: '',
      email: '',
      password: '',
      confirmPassword: '',
    };
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = async e => {
    e.preventDefault();

    const { displayName, email, password, confirmPassword } = this.state;
    const { signUpStart } = this.props;

    if (password !== confirmPassword) {
      alert('Hasła nie są takie same! Spróbuj ponownie.');
    } else {
      signUpStart(email, password, displayName);
    }
  };

  render() {
    const { displayName, email, password, confirmPassword } = this.state;

    return (
      <div className="sign-up">
        <h2 className="sign-up__title">Zarejestruj się</h2>
        <form>
          <FormInput
            type="text"
            name="displayName"
            label="Nazwa"
            value={displayName}
            onChange={this.handleChange}
            required
          />
          <FormInput
            type="email"
            name="email"
            label="Email"
            value={email}
            onChange={this.handleChange}
            required
          />
          <FormInput
            type="password"
            name="password"
            label="Hasło"
            value={password}
            onChange={this.handleChange}
            required
          />
          <FormInput
            type="password"
            name="confirmPassword"
            label="Potwierdź hasło"
            value={confirmPassword}
            onChange={this.handleChange}
            required
          />
          <div className="button-wrapper">
            <CustomButton handleClick={this.handleSubmit}>
              Zarejestruj
            </CustomButton>
          </div>
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  signUpStart: (email, password, displayName) =>
    dispatch(signUpStart({ email, password, displayName })),
});

export default connect(null, mapDispatchToProps)(SignUp);
