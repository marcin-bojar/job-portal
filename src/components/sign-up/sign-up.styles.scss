.sign-up {
  border: 2px solid #000;
  border-radius: 7px;
  padding: 3rem;
  height: fit-content;
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.2);

  &__title {
    margin-bottom: 5rem;
    font-size: 3rem;
  }
}

.button-wrapper {
  margin-top: 3.5rem;
}
