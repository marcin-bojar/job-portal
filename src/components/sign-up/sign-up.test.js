import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { SignUp } from './sign-up.component';
import CustomButton from '../custom-button/custom-button.component';
import FormInput from '../form-input/form-input.component';

const mockStore = configureStore([]);
let store = mockStore({});
store.dispatch = jest.fn();
let wrapper;
let props = {};

describe('SignUp component connected to redux store', () => {
  beforeEach(() => {
    props.signUpStart = jest.fn();
    wrapper = shallow(<SignUp store={store} {...props} />);
  });

  it('renders SignUp component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders 4 FormInput components', () => {
    expect(wrapper.find(FormInput).length).toEqual(4);
  });

  it('calls signUpStart function when sign up button is clicked', () => {
    const spy = jest.spyOn(wrapper.instance().props, 'signUpStart');
    mount(<SignUp store={store} {...props} />)
      .find(CustomButton)
      .simulate('click')
      .simulate('click');
    expect(spy).toHaveBeenCalledTimes(2);
  });

  it('handles form input change correctly', () => {
    let mockEvent = {
      target: {
        name: 'displayName',
        value: 'abc',
      },
    };

    const mockState = {
      displayName: '',
      email: '',
      password: '',
      confirmPassword: '',
    };
    wrapper.instance().setState(mockState);
    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state()).toEqual({
      displayName: 'abc',
      email: '',
      password: '',
      confirmPassword: '',
    });

    mockEvent = {
      target: {
        name: 'email',
        value: 'abc@gmail.com',
      },
    };

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state()).toEqual({
      displayName: 'abc',
      email: 'abc@gmail.com',
      password: '',
      confirmPassword: '',
    });
  });
});
