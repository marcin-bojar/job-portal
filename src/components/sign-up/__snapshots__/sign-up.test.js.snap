// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component connected to redux store renders SignUp component correctly 1`] = `
<div
  className="sign-up"
>
  <h2
    className="sign-up__title"
  >
    Zarejestruj się
  </h2>
  <form>
    <FormInput
      label="Nazwa"
      name="displayName"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <FormInput
      label="Email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <FormInput
      label="Hasło"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <FormInput
      label="Potwierdź hasło"
      name="confirmPassword"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <div
      className="button-wrapper"
    >
      <CustomButton
        handleClick={[Function]}
      >
        Zarejestruj
      </CustomButton>
    </div>
  </form>
</div>
`;
